analytics.js

```diff
@@ -1336,21 +1336,34 @@ const eventsPushedAlready =
   !!window.rudderanalytics &&
   window.rudderanalytics.push == Array.prototype.push;
 
-const argumentsArray = window.rudderanalytics;
-
-while (argumentsArray && argumentsArray[0] && argumentsArray[0][0] !== "load") {
-  argumentsArray.shift();
+const argumentsArray = window.rudderanalytics || [];
+/**
+ * Iterate the call stack until we find load call and
+ * then shift it to the beginning.
+ *
+ * Ex: Say the call stack currently have [page, identify, load, track]
+ * It will become [load, page, identify, track]
+ */
+let i = 0;
+while (i < argumentsArray.length) {
+  if (argumentsArray[i] && argumentsArray[i][0] === "load") {
+    argumentsArray.unshift(argumentsArray.splice(i, 1)[0]);
+    break;
+  }
+  i += 1;
 }
-if (
-  argumentsArray &&
-  argumentsArray.length > 0 &&
-  argumentsArray[0][0] === "load"
-) {
+
+if (argumentsArray.length > 0 && argumentsArray[0][0] === "load") {
   const method = argumentsArray[0][0];
   argumentsArray[0].shift();
   logger.debug("=====from init, calling method:: ", method);
   instance[method](...argumentsArray[0]);
   argumentsArray.shift();
+} else {
+  /**
+   * When load is not present in the call stack restrict other events from proceeding
+   */
+  argumentsArray.length = 0;
 }
```

Comment:
I redact my previous statement. Let's not remove the elements from the array in this scenario as I realized the situation would be the same for the query string API where they'd be waiting for the `load` API call.